ports/rp2/machine_uart.c

```diff
@@ -169,6 +252,36 @@ STATIC mp_obj_t machine_uart_make_new(const mp_obj_type_t *type, size_t n_args,
         self->invert = args[ARG_invert].u_int;
     }
 
+    self->read_lock = false;
+
+    // Set the buffer size and allocate the buffer
+    size_t rxbuf_len = DEFAULT_BUFFER_SIZE;
+    // get the length if set
+    if (args[ARG_rxbuf].u_int > 0) {
+        rxbuf_len = args[ARG_rxbuf].u_int;
+        // minimal size is DEFAULT_BUFFER_SIZE
+        // maximal size is MAX_BUFFER_SIZE
+        if (rxbuf_len < DEFAULT_BUFFER_SIZE) {
+            rxbuf_len = DEFAULT_BUFFER_SIZE;
```

Comment:
Is it needed to protect against small buffers?  What's wrong with a small buffer like 4 bytes?

Reply:
I had a version with a MIN_BUFFER_SIZE. I'll enable that. The smallest reasonbale size seems to the the FIFO size, which is 32.